File: 27-remove-element/remove-element.c
int removeElement(int* nums, int numsSize, int val) {

    int countElements = 0;

    for (int i = 0; i < numsSize; i++) {
        
        if (nums[i] != val){
            nums[countElements] = nums[i];
            countElements++;
        }
    }

    return countElements;

}

File: 27-remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        

        int size = nums.length;
        int writeIndex = 0;

        for (int i = 0; i < size; i++) {

            if (nums[i] != val) {

                nums[writeIndex] = nums[i];
                writeIndex++;
            }            

        }

        return writeIndex;
        

    }
}